packages/xo-web/src/xo-app/xoa/update/index.js

```diff
@@ -84,25 +89,52 @@ const Updates = decorate([
     initialState: () => ({
       ...initialProxyState(),
       ...initialRegistrationState(),
+      ...initialChannels(),
       askRegisterAgain: false,
       showPackagesList: false,
     }),
     effects: {
       async configure() {
-        await xoaUpdater.configure(
-          pick(this.state, [
+        const { effects, state } = this
+        await xoaUpdater.configure({
+          ...pick(state, [
             'proxyHost',
             'proxyPassword',
             'proxyPort',
             'proxyUser',
-          ])
-        )
-        return this.effects.resetProxyConfig()
+          ]),
+          channel: state.isUnlistedChannel
+            ? state.unlistedChannel
+            : state.channel,
+        })
+        await Promise.all([effects.resetProxyConfig(), effects.update()])
       },
-      initialize() {
-        return this.effects.update()
+      async initialize() {
+        await Promise.all([
+          this.effects.initializeChannels(),
+          this.effects.update(),
+        ])
+      },
+      async initializeChannels() {
+        const { state } = this
+
+        const channels = await xoaUpdater.getReleaseChannels()
```

Comment:
As i see, channels are not changed by the UI. 
To avoid, declaring a var then changing its value, you can  use a computed which returns a sorted array of channels.

Reply:
It won't be trivial because we are setting the state as a result of this method.

There is probably a better to do this but it might not be trivial, feel free to help @HamadaBrest with this.

Reply:
@julien-f mainly the var that @badrAZ is talking about will be cleaned by `javascript garbage collector` right ?

Reply:
@julien-f  
```js
computed: {
  channels: xoaUpdater.getReleaseChannels,
}
```
@HamadaBrest 
My idea is simply separate the contexts. One computed for channels, one computed for options and one computed for channel.

Reply:
@badrAZ  Be careful, `getReleaseChannels` is not a bound a method.

Also, how do you make sure the the select is correctly initialized in this case.
This is certainly possible but not trivial, test it on your side first.

Reply:
I think that we should not use `computed values` for something else.
a `computed` is a value derived from the state.

Reply:
@HamadaBrest While I though this too initially, I've come to expand this definition to any value that is dynamically computed (and should not be changed like a state value), which is the case here because it's the result of a function.